src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -1036,9 +1357,33 @@ Shard::valFetch(uint256 const& hash) const
     catch (std::exception const& e)
     {
         return fail(
-            std::string("exception ") + e.what() + " in function " + __func__);
+            std::string("Exception caught in function ") + __func__ +
+            ". Error: " + e.what());
     }
 }
 
+std::unique_ptr<Shard::Count>
+Shard::makeBackendCount()
```

Comment:
I like your suggestion of avoiding heap use. The sweep can be handled with an optional but I wonder, since `backendCount_` is atomic, should we simply access it without the lock? I haven't studied that scenario yet. Thoughts?

Reply:
One primary reason to use RAII is correctness in the face of exceptions.  If you just increment `backendCount_` you'll need to be sure the decrement happens regardless of whether an exception is thrown.  Probably the easiest way to have long-standing confidence in that would be to use the `std::optional`.  But you can study the situation and make your own call.

Reply:
I meant creating an RAII local.
eg.
```
    {
        std::lock_guard lock(mutex_);
        ...
    }

    Shard::Count const scopedCount(&backendCount_);
```

As I typed that I realized the reason it can't be done. The backend could be closed before the increment.  I'll go the optional route. Thanks @scottschurr !